samples/JwtSample/wwwroot/index.html

```diff
@@ -0,0 +1,104 @@
+﻿<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8" />
+    <title>SignalR JWT Sample</title>
+</head>
+<body>
+    <div id="log">
+
+    </div>
+</body>
+</html>
+<script type="text/javascript" src="lib/signalr-client/signalr-client.js"></script>
+<script>
+
+    function createLog(id) {
+        var log = document.getElementById('log');
+        var ul =  document.createElement('ul');
+        ul.id = 'log' + id;
+        log.appendChild(ul);
+    }
+
+    function appendLog(id, entry) {
+        var listId = document.getElementById('log' + id);
+        if (listId.children.length > 11) {
+            listId.removeChild(listId.children[1]);
+        }
+        var child = document.createElement('li');
+        child.innerText = entry;
+        listId.appendChild(child);
+    }
+
+    function get(url) {
+        return new Promise((resolve, reject) => {
+            var xhr = new XMLHttpRequest();
+            xhr.open('GET', url, true);
+            xhr.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
+            xhr.send();
+            xhr.onload = () => {
+                if (xhr.status >= 200 && xhr.status < 300) {
+                    resolve(xhr.response || xhr.responseText);
+                }
+                else {
+                    reject(new Error(xhr.statusText));
+                }
+            };
+
+            xhr.onerror = () => {
+                reject(new Error(xhr.statusText));
+            }
+        });
+    }
+
+    var tokens = {};
+
+    function refreshToken(id) {
+        var tokenUrl = 'http://' + document.location.host + '/generatetoken?user=' + id;
+        return get(tokenUrl)
+            .then(function (token) {
+                tokens[id] = token;
+            });
+    }
+
+    function runConnection(id, transportType) {
+        var connection;
+
+        refreshToken(id)
```

Comment:
You're refreshing before every connection? How often is that likely to be? These tokens are meant to last at least an hour.

Reply:
I need a token to start a connection. Then I refresh the token [every 20 secs](https://github.com/aspnet/SignalR/pull/1101/files/41cde6fc455d2ea87591ac1567ca4769368c0f2f#diff-2de970c6f616cd91f56bcb443e2e8367R84) because the server  creates tokens that [expire after 30 secs](https://github.com/aspnet/SignalR/pull/1101/files/41cde6fc455d2ea87591ac1567ca4769368c0f2f#diff-4179aa1ab03cf92976d053318cf3f436R79). 
I know that tokens generally live longer but this sample is supposed to show that the functionality being added is enough to update the token before it expires and hence prevent from SignalR connection failures caused by auth.

Reply:
I think I need to change id to userId to make it cleaner - I open 3 connections.